04.class/memo_db.js

```diff
@@ -0,0 +1,77 @@
+import sqlite3 from "sqlite3";
+import Memo from "./memo.js";
+
+class MemoDb {
+  constructor() {
+    this.db = new sqlite3.Database("memo.db");
+    this.changeDb(
+      "CREATE TABLE IF NOT EXISTS memos (id INTEGER PRIMARY KEY AUTOINCREMENT, content TEXT NOT NULL)",
+    );
+  }
+
+  loadMemos() {
+    return this.fetchAll("SELECT * FROM memos").then((rows) => {
+      if (rows.length === 0) console.log("No note");
+      return rows.map((row) => new Memo(row.id, row.content));
+    });
+  }
+
+  fetchAll(query, params = []) {
+    return new Promise((resolve, reject) => {
+      this.db.all(query, params, (err, rows) => {
+        if (err) {
+          reject(err);
+        } else {
+          resolve(rows);
+        }
+      });
+    });
+  }
+
+  fetchOne(query, params) {
+    return new Promise((resolve, reject) => {
+      this.db.get(query, params, (err, row) => {
+        if (err) {
+          reject(err);
+        } else {
+          const memo = new Memo(row.id, row.content);
+          resolve(memo);
+        }
+      });
+    });
+  }
+
+  changeDb(sql, params = []) {
```

Comment:
単にSQLを実行するメソッド、と見なして、

```suggestion
  execute(sql, params = []) {
```

と名付けた方が、ほどよく抽象的で再利用性が高くなりそうです。

https://bootcamp.fjord.jp/articles/78

Reply:
メソッド名を`changeDb()`→`execute()`に修正しました。